pkg/clients/gitlab.go

```diff
@@ -93,39 +87,12 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 		if err := c.Get(ctx, types.NamespacedName{Namespace: csr.Namespace, Name: csr.Name}, s); err != nil {
 			return nil, errors.Wrap(err, "cannot get credentials secret")
 		}
-		return UseProviderSecret(ctx, s.Data[csr.Key], DefaultSection)
+		return &Config{BaseURL: pc.Spec.BaseURL, Token: string(s.Data[csr.Key])}, nil
```

Comment:
`BaseURL` is not marked with `kubebuilder:validation:Required`, so it might be empty. It the provider able to deal with that or should it instead default to `https://gitlab.com`?

Reply:
We are indeed defaulting to gitlab.com in case the `baseURL` is empty: https://github.com/crossplane-contrib/provider-gitlab/blob/master/pkg/clients/gitlab.go#L51-L55

Reply:
(or rather the gitlab sdk is)

Reply:
OK, then this should resolve it

Reply:
Thanks! I just tested it as well and it works as expected 👍